opacclient/libopac/src/main/java/de/geeksfactory/opacclient/apis/Wise.kt

```diff
@@ -0,0 +1,563 @@
+            type = mediatypes[mediumSrt]
+
+        }
+    }
+
+
+    private fun getFieldValue(fieldObject: JSONObject?, key: String = "value", flatten: Boolean = false): String? {
+        if (fieldObject == null) return null
+
+        val contentObject = fieldObject.optJSONObject("content")
+        if (contentObject != null) {
+            return contentObject.getString(key)
+        }
+        val arr = fieldObject.optJSONArray("content")
+        if (arr != null && arr.length() > 0) {
+            if ( flatten) {
+                return  List(arr.length(), arr::getJSONObject).map{ o -> o.getString(key) }.reduce{ u,v -> "$u, $v"}
+            }
+            return arr.getJSONObject(0).getString(key)
+        }
+        return null
+    }
+
+    private fun getImgUrl(momkeys: String?, size: Int = 80): String? {
+        return "$baseurl${imgurl}size=${size}&$momkeys"
+    }
+
+    override fun search(query: List<SearchQuery>): SearchRequestResult {
+        mediaTypeQS= ""
+        vcgrpt = "0"
+        query.forEach { q -> when(q.searchField.id){
+            "materiaal"   -> mediaTypeQS= "&wf_medium_srt=${q.value}"
+            "search"      -> queryString = q.value.split(" ").reduce { u, v -> u + "%20" + v }
+            "searchScope" -> scope = q.value
+            "location"    -> vcgrpt = q.value
+        } }
+        return searchGetPage(1)
+    }
+
+    override fun searchGetPage(pagina: Int): SearchRequestResult {
+        val AMOUNT = 10
+        val page = pagina -1
+        val response = httpPost(
+                "$baseurl/cgi-bin/bx.pl",
+                ("prt=INTERNET&var=portal&vestnr=$homeBranch&fmt=json&search_in=$scope&amount=$AMOUNT&catalog=default&event=osearch" +
+                        "&preset=all&offset=${page*AMOUNT}$mediaTypeQS&qs=$queryString&vcgrpf=0&backend=wise&vcgrpt=$vcgrpt").toRequestBody("application/x-www-form-urlencoded".toMediaTypeOrNull()),
+                ENCODING, false)
+
+        val json = JSONObject(response)
+        val objects = List(json.getJSONArray("objects").length(), json.getJSONArray("objects")::getJSONObject)
+        val searchResultList = objects.map { o -> o.getJSONObject("fields") }.filter { fields -> getFieldValue(fields.getJSONObject("medium_srt")) != "Website" } .map { fields -> parseSearchResult(fields) }
+
+        with(json.getJSONObject("paging")) {
+            return SearchRequestResult(
+                    searchResultList,
+                    getInt("total"),
+                    getInt("total") / getInt("perpage"),
+                    getInt("offset") / getInt("perpage")
+            )
+        }
+    }
+
+
+    override fun cancel(media: String?, account: Account?, useraction: Int, selection: String?): OpacApi.CancelResult {
+        val itemId = media!!.split(":")[2]
+
+        httpDelete("$baseurl/restapi/patron/${auth!!.patronSystemId}/library/${auth!!.libraryId}/hold/$itemId")
+        return OpacApi.CancelResult(OpacApi.MultiStepResult.Status.OK)
```

Comment:
We should not return OK without any check if this actually worked, maybe at least check the HTTP response code?